File: python/docmost/services/export_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import ExportPageDto, ExportSpaceDto


class ExportApi:
    """Export API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def export_page(self, request: ExportPageDto) -> ApiResponse[Any]:
        return self._http.post("pages/export", body=request.model_dump(by_alias=True, exclude_none=True))

    def export_space(self, request: ExportSpaceDto) -> ApiResponse[Any]:
        return self._http.post("spaces/export", body=request.model_dump(by_alias=True, exclude_none=True))

File: go/docmost/services/export_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// ExportAPI implements Export API.
type ExportAPI struct {
	http *httpclient.HTTPClient
}

func NewExportAPI(http *httpclient.HTTPClient) *ExportAPI {
	return &ExportAPI{http: http}
}

func (a *ExportAPI) ExportPage(ctx context.Context, request *models.ExportPageDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "pages/export", request)
}

func (a *ExportAPI) ExportSpace(ctx context.Context, request *models.ExportSpaceDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "spaces/export", request)
}

